### DISCUSSION TITLE
Using Cauchy Stress as Target Value in Homogenization System

### DISCUSSION THREAD

**Discussion by grld** [idaholab/moose#23275] [Q&A Modules: Solid mechanics] [answered]
Dear community,

I would like to use the MOOSE tensor mechanics module to run stress-controlled RVE calculations. I read about the `homogenization system` and that it can be used in combination with periodic boundary conditions to impose cell-average stress or strain conditions. 

The documentation states that in the finite strain framework, either the deformation gradient **F** or the first Piola-Kirchhoff **PK1** can be used as target quantities. For the calculations that I would like to run, I am interested in targeting the Cauchy stress tensor **and** use the finite strain formulation. The reason is that I don't know the deformation gradient F _before_ I run the simulation so I can not define the **PK1** beforehand as boundary condition but only the cauchy.

I know that this is theoretically possible in Abaqus as explained in this [work](https://hss-opus.ub.ruhr-uni-bochum.de/opus4/frontdoor/index/index/docId/5082). The author implements periodic boundary conditions by coupling edge and face nodes at opposite sides of the RVE. Either Neumann or Dirichlet boundary conditions are applied on so-called reference nodes at the corners of the RVE (see page 27). Defining Neumann boundary conditions on the reference nodes is then equivalent to defining an average Cauchy stress (see page 80)

I was wondering if this can be also done in MOOSE. Has someone experience with this or any recommendations?

Thank you very much for your support already:)

**Comment by ngrilli** [comment, marked as answer]
@grld  you should be able to apply periodic boundary conditions on the displacement variables,
examples of periodic boundary conditions tests are here:
https://github.com/idaholab/moose/tree/next/test/tests/bcs/periodic

**Reply by hugary1995** [reply]
yeah, that's one way to impose KUBC (well PBC strictly speaking, which is a mix of KUBC and SUBC) but probably isn't what the OP is after. Right now we support SUBC with given homogenized PK1 stress. Adding support for Cauchy stress should be possible and is a good addition IMHO. Feel free to open a feature request and tag me there.

**Reply by grld** [reply]
Hello, thank you very much for your answers. 
@ngrilli I am aware of the possibility to apply periodic boundary conditions (PBC) in Moose together with Dirichlet BC. As @hugary1995 mentions there, is the possibility to combine them with homogenized PK1 stress (in case of finite strains) as explained here: https://mooseframework.inl.gov/modules/tensor_mechanics/Homogenization.html 
Adding the Cauchy stress as the homogenized or targeted quantity there would be exactly what I am looking for.

@hugary1995 I have never opened a feature request before. Is there any guide to this?

Thanks to both of you again:)

**Reply by hugary1995** [reply]
You can open a new issue and then choose "feature request" (rather than bug report). I've done that for you: https://github.com/idaholab/moose/issues/23287
